test/tools/ossfuzz/protoToSol.h

```diff
@@ -0,0 +1,185 @@
+ *      // a non-zero error code for failure, a zero uint
+ *      // when all tests pass
+ *      function test() public pure returns (uint)
+ *      {
+ *          C0 tc0 = new C0();
+ *          if (tc0.f0() != 1337)
+ *              return 1;
+ *          C0B tc1 = new C0B();
+ *          if (tc1.f0() != 42)
+ *              return 2;
+ *          // Expected return value if all tests pass
+ *          return 0;
+ *      }
+ * }
+ */
+class ProtoConverter
+{
+public:
+	ProtoConverter() {}
+	ProtoConverter(ProtoConverter const&) = delete;
+	ProtoConverter(ProtoConverter&&) = delete;
+	std::string protoToSolidity(Program const&);
+	/// @returns true if test calls a library function, false
+	/// otherwise
+	bool libraryTest() const;
+	/// @returns name of the library under test
+	std::string libraryName() const;
+private:
+	/// Variant type that points to one of contract, interface, library protobuf messages
+	using CIL = std::variant<Contract const*, Interface const*, Library const*>;
+	/// Protobuf message visitors that accept a const reference to a protobuf message
+	/// type and return its solidity translation.
+	std::string visit(Program const&);
+	std::string visit(TestContract const&);
+	std::string visit(ContractType const&);
+	std::string visit(Interface const& _interface);
+	std::string visit(Library const& _library);
+	std::string visit(Contract const& _contract);
+	/// @returns a string pair containing a library declaration (relevant for library
+	/// tests only) and a solidity test case
+	std::pair<std::string, std::string> generateTestCase(TestContract const& _testContract);
+	/// @returns name of a program i.e., contract, library or interface
+	std::string programName(CIL _program);
+	/// @returns a tuple containing the names of the library and function under
+	/// test, and its expected output.
+	std::tuple<std::string, std::string, std::string> pseudoRandomLibraryTest();
+	/// Performs bookkeeping for a fuzzer-supplied program
+	void openProgramScope(CIL _program);
+	/// @returns a deterministic pseudo random unsigned integer
+	unsigned randomNumber();
+	/// @returns true if fuzzer supplied Library protobuf message
+	/// contains zero functions, false otherwise.
+	static bool emptyLibrary(Library const& _library)
+	{
+		return _library.funcdef_size() == 0;
+	}
+	/// @returns true if there are no valid library test cases, false
+	/// otherwise.
+	bool emptyLibraryTests()
+	{
+		return m_libraryTests.size() == 0;
+	}
+	/// @returns true if there are no valid contract test cases, false
+	/// otherwise.
+	bool emptyContractTests()
+	{
+		return m_contractTests.size() == 0;
+	}
+	/// Numeric suffix that is part of program names e.g., "0" in "C0"
+	unsigned m_programNumericSuffix = 0;
+	/// Flag that states whether library call is tested (true) or not (false).
+	bool m_libraryTest = false;
+	/// A smart pointer to fuzzer driven random number generator
+	std::shared_ptr<SolRandomNumGenerator> m_randomGen;
```

Comment:
Why not `unique_ptr`?
Or why pointer at all?

Reply:
Not unique_ptr because it will be shared with an upcoming adaptor class that also makes use of it. Pointer because passing the struct as is felt heavy when a pointer to it would work just fine.